### DISCUSSION TITLE
C-TAEA

### DISCUSSION THREAD

**Discussion by RafaHPSUnicamp** [anyoptimization/pymoo#517] [General]
Hi, I am revising my work related to C-TAEA and other algoritms. So, it is really common that C-TAEA can select points to enter in Pareto Front that disobey certain restrictions? For example, my C-TAEA have points which variables have values lower as their restrictions supposed to not allow. It is normal?

**Comment by cyrilpic** [comment]
C-TAEA is a dual population algorithm. One population enforces constraints, and the other does not. Then, there is a mechanism to pass solutions between the two. It is normal to see solutions not respecting constraints in the population for that reason. This algorithm works best if the Pareto front of the constrained problem is close (including in the decision space) to the Pareto front of the unconstrained problem.

**Reply by RafaHPSUnicamp** [reply]
Hi, so there is two Pareto Fronts from both populations? And there is possible to plot the Pareto Front from each one, or only from both, specially in Pymoo?

**Reply by cyrilpic** [reply]
You can get both populations:
`algorithm.pop, algorithm.da`

DA is the diversity archive

**Reply by RafaHPSUnicamp** [reply]
I see. Can you exemplify in a simple code? If possible. Maybe I modify my code, but I need to know how to apply algorithm.pop and algorithm.da in code, since I learned to use Pymoo through examples...

**Reply by cyrilpic** [reply]
It's hard to provide an example since these are internal variables of the algorithm. There is no real use case. If you want to plot their content, you can access them from the results structure:
```
res = minimize(...)
pop = res.algorithm.pop
da = res.algorithm.da
```
And then your standard plotting code (pop.get('F') or da.get('F') and so on).
That being said, I'm not sure this will provide much value to you. C-TAEA is probably a good algorithm for only very few _real-world problems_.

**Reply by cyrilpic** [reply]
If you want to dig deeper into this algorithm, you should absolutely read the related paper.
